### PR TITLE
Invoke inst_update_installer client after system_analysis

### PR DESCRIPTION
**Pull request by imobach**
https://fate.suse.com/319716
https://trello.com/c/T49hdvDd/498-5-self-update-change-dud-support-to-rpmmd-repository-with-rpms
https://github.com/yast/yast-installation/pull/356

The self-update client should be invoked from control file.

### CODE REVIEW

File: control/control.SLES.xml [hunk 1]
```diff
@@ -729,6 +729,12 @@ Please visit us at http://www.suse.com/.
                     <enable_next>yes</enable_next>
                 </module>
                 <module>
+                    <label>Installer Update</label>
+                    <name>update_installer</name>
+                    <enable_back>yes</enable_back>
+                    <enable_next>yes</enable_next>
```
Comment (mvidner) [inline control/control.SLES.xml]: The default is to have Back and Next enabled, I have verified it with the System Role step recently. But it is not documented :-/

**Comment by mvidner** [comment]
For the .changes file:
> Automatically update the installer in the initial stage of: installation, update, autoupgrade. (Not affected: autoinstallation, live_installation) FATE#319716.

I do not see why autoupgrade should be included AND autoinstallation excluded. It seems inconsistent. But I don't know which is better.

**Comment by imobach** [comment]
It's my fault: I think that autoinstallation should be included but I missed it because no system_analysis is performed. I guess we need to find a point in autoinstallation where the software manager is initialized (like system_analysis does).

**Comment by imobach** [comment]
Also I don't know what's the reason to skip the system analysis in regular autoinstallation. I have to investigate.

**Comment by mvidner** [comment]
LGTM.
What about https://github.com/yast/skelcd-control-openSUSE and https://github.com/yast/skelcd-control-SLED ? SLED is explicitly requested in FATE and openSUSE has been mentioned but no decision reached whether it is wanted.

**Comment by imobach** [comment]
@mvidner Well, I made a mistake and I've added the call too early in almost all cases (except during the regular installation). So I'm updating the PR. Sorry.

**Comment by mvidner** [comment]
> added the call too early

Ah, I've missed that. So the reason for the position is that it should be rather soon but not before the packager is initialized which is in system_analysis? We could add that as a comment to the `<module>` element.

**Comment by imobach** [comment]
Yes, it makes sense. I've added a comment before the `<module>` element. Thanks!

**Comment by mvidner** [comment]
LGTM, thanks!

**Comment by imobach** [comment]
Thank you for the review!

**Pull request merged**